R/utilities-pksim.R

```diff
@@ -0,0 +1,162 @@
+#'
+#' @note  This will only work on Windows machine and should not be called on any other OS.
+#' This function should also only be called explicitely when using a portable install of the package
+#'
+#' @import rClr
+#' @export
+initPKSim <- function(pksimFolderPath = NULL) {
+  if(ospsuiteEnv$isPKSimLoaded){
+    return()
+  }
+
+  pksimFolderPath <- pksimFolderPath %||% .getPathToPKSimInstallDir()
+  .addPathToSystemPath(pksimFolderPath)
+  pksimR <- file.path(pksimFolderPath, "PKSim.R.dll")
+  if (!file.exists(pksimR)) {
+    stop(messages$pkSimRPathInvalid(pksimR))
+  }
+  rClr::clrLoadAssembly(pksimR)
+  rClr::clrCallStatic("PKSim.R.Api", "InitializeOnce")
+
+  # Only set the flag if initialization was successful
+  ospsuiteEnv$isPKSimLoaded <- TRUE
+  print(ospsuiteEnv$isPKSimLoaded)
+}
+
+
+#' Tries to find the installation path for a specific version of PK-Sim from an Windows registry entry.
+#'
+#' @param pksim.version The version number of Pk-Sim as a string.
+#'
+#' @return
+#' The path to the PK-Sim installation for version pksim.version or NA if no path could be found.
+#' The path is separated with slashes (unix-style) and in compilance with base-R without a trailing slash.
+#'
+#' @examples
+#' path <- .getPathToPKSimInstallDirFromRegistry("7.4")
+.getPathToPKSimInstallDirFromRegistry <- function(pksim.version) {
+  pksimVersion <- trimws(pksim.version)
+
+  if (.Platform$OS.type != "windows") {
+    stop("Only Windows platforms are supported")
+  }
+
+  suite.name <- ospsuiteEnv$suiteName
+  product.name <- "PK-Sim"
+  reg.path <- file.path("SOFTWARE",
+    suite.name,
+    product.name,
+    pksim.version,
+    fsep = "\\"
+  )
+
+  reg.entry <- NA
+  try(reg.entry <- readRegistry(reg.path, hive = "HLM", maxdepth = 1, view = "64-bit"),
+    silent = TRUE
+  )
+
+  if ("InstallDir" %in% names(reg.entry)) {
+    return(.homogenizePath(reg.entry$InstallDir))
+  }
+
+  return(NA)
+}
+
+#' Tries to find the installation path for a specific version of PK-Sim via the filesystem.
+#' Searching is done in the following order:
+#' 1. Search via filesystem in a guessed installation folder from the base.search.folder
+#' 3. Search via filesystem for PKSim.exe recursivly from the defined base.search.folder (fallback)
+#'
+#' @param pksim.version The version number of Pk-Sim as a string.
+#' @param base.search.folder The base folder for filesystem-lookup fallback (default: 64-bit program folder)
+#'
+#' @return
+#' The path to the PK-Sim installation for version pksim.version or NA if no path could be found.
+#' The path is separated with slashes (unix-style) and in compilance with base-R without a trailing slash.
+#' If more than one matching path is found a warning is produced.
+#'
+#' @examples
+#' path <- .getpathToPKSimInstallDirFromFileSystem("7.4")
+#' path2 <- .getpathToPKSimInstallDirFromFileSystem("7.5", "C:/MyOSPFolder/")
+.getpathToPKSimInstallDirFromFileSystem <- function(pksim.version,

```

Comment:
@PavelBal @Yuri05  Any idea why we need this? This is absolutely ridiculous to not use the registry. Or is this because of possible issue with acccess rights?
If pksim is installed properly, this value is in registry. Otherwise, there are serious problem with the install anyways

I have copied this code from the current R toolbox but I feel that this is not required